tests/test_basic.py

```diff
@@ -16,105 +19,162 @@ def test_show_config():
 
 
+@pytest.mark.skipif(platform.system() == "Windows", reason="uname command will not run on Windows.")
 def test_uname_m():
     out = subprocess.check_output(["uname", "-m"])
     print(out.decode("utf-8"))
 
 
 def test_install_dependencies():
-    with tempfile.TemporaryDirectory() as tempdir:
-        mci = MinicondaInstaller(tempdir, verbose=True)
-        mci.download()
-        mci.install_miniconda()
-        mci.update_miniconda_base()
-        mci.install_pip(dependencies=["geopy"])
-        mci.install_conda(channel="conda-forge", dependencies=["pyyaml"])
-
-        # Run Miniconda's Python and import the installed dependencies.
-        for pkg_name in ["geopy", "pyyaml"]:
-            if pkg_name == "pyyaml":
-                pkg_name = "yaml"
-            cmd = [
-                "%s/bin/python" % tempdir,
-                "-c",
-                '''import %s; print("%s %%s ok" %% %s.__version__)''' % \
-                    (pkg_name, pkg_name, pkg_name)
-            ]
-            out = subprocess.check_output(cmd).decode("utf-8").strip()
-            print(out)
-            assert re.match("%s .+ ok" % pkg_name, out)
+    try:
+        with tempfile.TemporaryDirectory() as tempdir:
+            mci = MinicondaInstaller(tempdir, verbose=True)
+            mci.download()
+            mci.install_miniconda()
+            mci.update_miniconda_base()
+            mci.install_pip(dependencies=["geopy"])
+            mci.install_conda(channel="conda-forge", dependencies=["pyyaml"])
+
+            if platform.system() == "Windows":
+                py_exe = "%s\\python" % tempdir
+            else:
+                py_exe = "%s/bin/python" % tempdir
+
+            # Run Miniconda's Python and import the installed dependencies.
+            for pkg_name in ["geopy", "pyyaml"]:
+                if pkg_name == "pyyaml":
+                    pkg_name = "yaml"
+                if platform.system() == "Windows":
+                    cmd = [
+                        "%s\\condabin\\activate"%(tempdir),
+                        "&&",
+                        py_exe,
+                        "-c",
+                        '''import %s; print("%s %%s ok" %% %s.__version__)''' % \
+                            (pkg_name, pkg_name, pkg_name)
+                    ]
+                else:
+                    cmd = [
+                        py_exe,
+                        "-c",
+                        '''import %s; print("%s %%s ok" %% %s.__version__)''' % \
+                            (pkg_name, pkg_name, pkg_name)
+                    ]
+
+                if platform.system() == "Windows":
+                    out = subprocess.check_output(cmd, shell=True).decode("utf-8").strip()
+                else:
+                    out = subprocess.check_output(cmd).decode("utf-8").strip()
+                print(out)
+                if platform.system() == "Windows":
+                    assert re.search("%s .+ ok" % pkg_name, out)
```

Comment:
Why is this `re.search` for Windows?

Reply:
because on windows I had to activate conda environment and import in same process. as re.match searches at the start of the string. hence used re.search in windows as output string is conda activate and then package.__version__ ok